pkg/backup/resource_backupper.go

```diff
@@ -276,6 +285,80 @@ func (rb *defaultResourceBackupper) backupResource(group *metav1.APIResourceList
 	return nil
 }
 
+// Adds CRD to the backup if one is found corresponding to this resource
+func (rb *defaultResourceBackupper) backupCRD(
+	log logrus.FieldLogger,
+	gr schema.GroupResource,
+	gv schema.GroupVersion,
+	itemBackupper ItemBackupper,
+) {
+	crdGr := schema.GroupResource{Group: apiextv1beta1.GroupName, Resource: "customresourcedefinitions"}
+	crdClient, err := rb.dynamicFactory.ClientForGroupVersionResource(apiextv1beta1.SchemeGroupVersion,
+		metav1.APIResource{
+			Name:       "customresourcedefinitions",
+			Namespaced: false,
+		},
+		"",
+	)
+	if err != nil {
+		log.WithError(err).Error("Error getting dynamic client for CRDs")
+		return
+	}
+
+	unstructuredCRDList, err := crdClient.List(metav1.ListOptions{FieldSelector: fmt.Sprintf("metadata.name=%s", gr.String())})
+	if err != nil {
+		log.WithError(errors.WithStack(err)).Error("Error listing CRDs")
+		return
+	}
+	crds, err := meta.ExtractList(unstructuredCRDList)
+	if err != nil {
+		log.WithError(errors.WithStack(err)).Error("Error extracting CRD list")
+		return
+	}
+	for _, item := range crds {
+		unstructured, ok := item.(runtime.Unstructured)
+		if !ok {
+			log.Errorf("Unexpected type %T", item)
+			continue
+		}
+
+		crd := apiextv1beta1.CustomResourceDefinition{}
+		itemMarshal, _ := json.Marshal(unstructured)
+		json.Unmarshal(itemMarshal, &crd)
```

Comment:
probably want to use `runtime.DefaultUnstructuredConverter.FromUnstructured` here instead of marshalling/unmarshalling

Reply:
I noticed after I submitted this PR that Velero was using the `DefaultUnstructuredConverter` elsewhere. I had pulled the marshal/unmarshal pattern from some of our plugin code which tends to use it inside backup/restore plugins for converting the item. I'll modify this to match what you're doing elsewhere with the `DefaultUnstructuredConterter`.

Reply:
Even better, as I'm pulling out the CRD version checking code, I can go back to just working with the Unstructured resource object, so I don't need to worry about converting it with the next iteration of the code.